### PR TITLE
github-electron: add power-monitor module events

### PR DESCRIPTION
**Pull request by miniak**

### CODE REVIEW

**Comment by jedmao** [comment]
:+1:

File: github-electron/github-electron.d.ts [hunk 1]
```diff
@@ -1823,6 +1823,30 @@ declare namespace Electron {
...
+		on(event: 'resume', listener: Function): this;
+		/**
+		 * Emitted when the system changes to AC power.
+		 */
+		on(event: 'on-ac', listener: Function): this;
+		/**
+		 * Emitted when system changes to battery power.
+		 */
+		on(event: 'on-battery', listener: Function): this;
+		on(event: string, listener: Function): this;
```
Comment (rhysd) [inline github-electron/github-electron.d.ts]: This method is needed?  If allowing `string` to `event` param, I think above string literal types does mean nothing.

Comment (miniak) [inline github-electron/github-electron.d.ts]: Unfortunatelly, that's how overloading based on value works in TypeScript. You must have the generic case

Comment (rhysd) [inline github-electron/github-electron.d.ts]: Ah, I got it... :sob:

**Comment by rhysd** [comment]
:+1:

**Comment by miniak** [comment]
Superseded by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8723

**Pull request closed without merging**